src/main/kotlin/me/shedaniel/linkie/namespaces/LegacyYarnNamespace.kt

```diff
@@ -1,96 +1,57 @@
-                            repo = legacyFabricMaven,
-                            showError = false
-                        )
+                versions { legacyYarnBuilds.keys }
+                uuid { version ->
+                    legacyYarnBuilds[version]!!.maven.let { it.substring(it.lastIndexOf(':') + 1) }
+                }
+                mappings {
+                    MappingsContainer(it, name = "Legacy Yarn").apply {
+                        loadIntermediaryFromMaven(version, legacyFabricMaven, legacyFabricIntermediary)
+                        val yarnMaven = legacyYarnBuilds[version]!!.maven
+                        mappingsSource = loadNamedFromMaven(yarnMaven.substring(yarnMaven.lastIndexOf(':') + 1),
+                                showError = false, repo = legacyFabricMaven, group = legacyFabricYarn)
                     }
                 }
             }
         }
     }
 
-    override fun getAllVersions(): Sequence<String> = workingLegacyFabricVersions.asSequence() + singleSequenceOf("1.2.5")
-    override fun getDefaultLoadedVersions(): List<String> = listOf()
+    override fun getAllVersions(): Sequence<String> = legacyYarnBuilds.keys.asSequence()
+    override fun getDefaultLoadedVersions(): List<String> {
+        return latestLegacyYarnVersion?.let(::listOf) ?: listOf()
+    }
     override val defaultVersion: String
-        get() = latestLegacyFabricVersions
+        get() = latestLegacyYarnVersion!!
 
     override fun supportsMixin(): Boolean = true
     override fun supportsAW(): Boolean = true
 
     override suspend fun reloadData() {
-        val pom189 = URL("$legacyFabricMaven/net/fabricmc/yarn/maven-metadata.xml").readText()
-        SAXReader().read(pom189.reader()).rootElement
-            .element("versioning")
-            .element("versions")
-            .elementIterator("version")
-            .asSequence()
-            .map { it.text }
-            .groupBy { it.substringBefore('+') }
-            .forEach { (mcVersion, builds) ->
-                if (legacyFabricVersions.containsKey(mcVersion)) {
-                    legacyFabricVersions[mcVersion] = builds.last()
-                }
-            }
-        workingLegacyFabricVersions.clear()
-        workingLegacyFabricVersions.addAll(legacyFabricVersions.asSequence()
-            .filter { it.value != null }
-            .map { it.key }
-        )
-        latestLegacyFabricVersions = workingLegacyFabricVersions.maxByOrNull { it.toVersion() }!!
+        val buildMap = LinkedHashMap<String, MutableList<YarnNamespace.YarnBuild>>()
+        json.decodeFromString(ListSerializer(YarnNamespace.YarnBuild.serializer()), URL("https://meta.legacyfabric.net/v2/versions/yarn").readText())
+                .forEach { buildMap.getOrPut(it.gameVersion) { mutableListOf() }.add(it) }
+        buildMap.forEach { (version, builds) -> builds.maxByOrNull { it.build }?.apply { legacyYarnBuilds[version] = this } }
```

Comment:
Reset legacyYarnBuilds first before this line?

Reply:
Somehow it's not done in any Namespace...